src/internal/operations/backup.go

```diff
@@ -102,10 +106,36 @@ type detailsWriter interface {
 // ---------------------------------------------------------------------------
 
 // Run begins a synchronous backup operation.
-func (op *BackupOperation) Run(ctx context.Context) (err error) {
+func (op *BackupOperation) Run(ctx context.Context) error {
 	ctx, end := D.Span(ctx, "operations:backup:run")
-	defer end()
+	defer func() {
```

Comment:
@ryanfkeepers  - It's not obvious to me why we split `Run` into `Run` and `do`. Could you explain a bit please?

Reply:
Nevermind - I assume this is so that we "wrap" the entire execution in a single place and have a success/failure error message?

Reply:
Yeah, just a bit of cleanliness/readability.  May not be useful in the log run, we can always fold it back together.